### PR TITLE
Add support for automatic opencontainer labels

### PR DESCRIPTION
**Pull request by codrut-fc**
Adding support for automatically setting `opencontainers` labels as default. Also, now there is a way to turn off automatic labeling.

It's documented here:
* https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys

Using `-e PLUGIN_AUTO_LABEL=false`:

```sh
docker run --rm \
  -e PLUGIN_AUTO_LABEL=false  \
  -e PLUGIN_TAG=latest  \
  -e PLUGIN_REPO=octocat/hello-world  \
  -e DRONE_COMMIT_SHA=d8dbe4d94f15fe89232e0402c6e8a0ddf21af3ab \
  -v $(pwd):$(pwd) \
  -w $(pwd) \
  --privileged \
  plugins/docker --dry-run

+ /usr/local/bin/docker build \
  --rm=true \
  -f Dockerfile \
  -t d8dbe4d94f15fe89232e0402c6e8a0ddf21af3ab \
  . \
  --pull=true
```

With `-e PLUGIN_AUTO_LABEL=true`, `DRONE_REMOTE_URL`, and `DRONE_REPO_LINK`:

```sh
docker run --rm \
  -e PLUGIN_AUTO_LABEL=true \
  -e DRONE_REMOTE_URL=https://github.com/octocat/hello-world.git \
  -e DRONE_REPO_LINK=https://github.com/octocat/hello-world \
  -e PLUGIN_TAG=latest \
  -e PLUGIN_REPO=octocat/hello-world \
  -e DRONE_COMMIT_SHA=d8dbe4d94f15fe89232e0402c6e8a0ddf21af3ab \
  -v $(pwd):$(pwd) \
  -w $(pwd) \
  --privileged \
  plugins/docker \
  --dry-run

+ /usr/local/bin/docker build \
  --rm=true \
  -f Dockerfile \
  -t d8dbe4d94f15fe89232e0402c6e8a0ddf21af3ab \
  . \
  --pull=true \
  --label org.opencontainers.image.created=2021-01-11T17:28:58Z \
  --label org.opencontainers.image.revision=d8dbe4d94f15fe89232e0402c6e8a0ddf21af3ab \
  --label org.opencontainers.image.source=https://github.com/octocat/hello-world.git \
  --label org.opencontainers.image.url=https://github.com/octocat/hello-world
```

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

**Review by tboerger — APPROVED** [review]

**Review by robertstettner — APPROVED** [review]

**Comment by bradrydzewski** [comment]
lets simplify things by removing the deprecated labels entirely and only using the new labels going forward.  I would then recommend we use a boolean flag to disable automatic labeling, perhaps `auto_label: true|false` where the value defaults to true (since this is the current behavior).

**Comment by codrut-fc** [comment]
All don @bradrydzewski

**Comment by codrut-fc** [comment]
You might want to restart CI, looks like it reached docker hub's api rate limit...

File: docker.go [hunk 1]
```diff
@@ -252,19 +254,22 @@ func commandBuild(build Build) *exec.Cmd {
...
+		labelSchema := []string{
+			fmt.Sprintf("created=%s", time.Now().Format(time.RFC3339)),
+			fmt.Sprintf("revision=%s", build.Name),
+			fmt.Sprintf("source=%s", build.Remote),
+			fmt.Sprintf("url=%s", build.Link),
+		}
+		labelPrefix := "org.opencontainers.image"
 
-	if len(build.LabelSchema) > 0 {
-		labelSchema = append(labelSchema, build.LabelSchema...)
```
Comment (bradrydzewski) [inline docker.go:263]: @tboerger can you confirm that `label_schema` values should be ignored if the user sets `auto_label: false`?  This operation is wrapped by `if build.AutoLabel { ...` and I was wondering if it should be moved outside this block, or if this is fine as-is.

Comment (tboerger) [reply docker.go:263]: Adding an option to opt-out is fine, I know some people that really asked for that.

**Comment by robertstettner** [comment]
Friendly bump 😄

**Comment by robertstettner** [comment]
@bradrydzewski Could you please merge this one?

**Pull request merged**